src/Core.tsx

```diff
@@ -226,15 +239,36 @@ export class Core extends React.Component<IModalProps, IModalState> {
       .catch()
   }
 
-  /** Pre-Step 1 - user picked a wallet and waiting to connect */
-  private connectToWallet () {
+  /**
+   * Check if the provider supports multiple networks and if the user
+   * should choose a network first.
+   * @param provider The provider selected by the user to use
+   */
+  private checkIfMultipleNetworks (provider: RLoginIProviderUserOptions) {
+    const { rpcUrls } = this.props
+    // choose the network first:
+    if (['Ledger', 'Trezor', 'Dcent'].includes(provider.name) && rpcUrls) {
```

Comment:
Should this be configurable?

Reply:
I don't think so. This "choose network" feature is only compatible with our hardware providers since it passes the URL and chain Id in a [specific format](https://github.com/rsksmart/rLogin/pull/150/files#diff-f0006202dbed07b20ed558041889df8f0bbce2825b1c0a84bd44728aef0a33a2R187-R190). As such, we can't apply this directly to Portis as they send the chainId and URL in a [network](https://github.com/rsksmart/rLogin/blob/develop/sample/front/index.html#L95-L98) variable.

If the developer does not want to show this modal and only use on the configuration in the initial rLogin setup they can choose not to pass the rpcUrl parameter. Which is the second condition in the if statement above. This is explained in the readme.